### PR TITLE
Even more clang tidy fixes

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Source/gravity/Gravity.cpp [hunk 1]
```diff
@@ -3574,9 +3574,9 @@ Gravity::update_max_rhs()
             coeffs[lev].resize(AMREX_SPACEDIM);
 
             for (int i = 0; i < AMREX_SPACEDIM ; i++) {
-                coeffs[lev][i].reset(new MultiFab(amrex::convert(grids[lev],
-                                                  IntVect::TheDimensionVector(i)),
-                                                  dmap[lev], 1, 0));
+                coeffs[lev][i] = std::make_unique<MultiFab>(amrex::convert(grids[lev],
+                                                                           IntVect::TheDimensionVector(i)),
```
Comment (yut23) [inline Source/gravity/Gravity.cpp:3578]: Too much indentation

Comment (zingale) [reply Source/gravity/Gravity.cpp:3578]: I think it's correct, since that is an argument of `amrex::convert()`

Comment (yut23) [reply Source/gravity/Gravity.cpp:3578]: Oh, my mistake.

File: Source/driver/Castro_io.cpp [hunk 2]
```diff
@@ -1172,7 +1172,7 @@ Castro::plotFileOutput(const std::string& dir,
 #ifndef TRUE_SDC
     if (store_burn_weights) {
         MultiFab::Copy(plotMF, getLevel(level).burn_weights, 0, cnt, static_cast<int>(Castro::burn_weight_names.size()), 0);
-        cnt += static_cast<int>(Castro::burn_weight_names.size());
+        cnt += static_cast<int>(Castro::burn_weight_names.size());  // NOLINT
```
Comment (yut23) [inline Source/driver/Castro_io.cpp:1175]: It would be good to specify the check we're suppressing here: `// NOLINT(clang-analyzer-deadcode.DeadStores)`

Comment (zingale) [reply Source/driver/Castro_io.cpp]: indeed, that's nice.

Comment (zingale) [reply Source/driver/Castro_io.cpp]: updated

**Review by yut23 — APPROVED** [review]

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2023-03-14-001/index.html

**Pull request merged**